Blockzilla/OverlayView.swift

```diff
@@ -184,115 +168,98 @@ class OverlayView: UIView {
+            
+            if (hasSuggestionInIndex) {
+                self.setAttributedButtonTitle(
+                    phrase: self.searchSuggestions[index],
+                    button: self.searchButtonGroup[index],
+                    localizedStringFormat: Settings.getToggle(.enableSearchSuggestions) ? "" : UIConstants.strings.searchButton
+                )
             }
         }
     }
 
-    private func updateFindInPagePlacement() {
+    fileprivate func updateFindInPageConstraints(isFindInPageHidden: Bool, lastSearchButtonIndex: Int) {
+        findInPageButton.isHidden = isFindInPageHidden
+        
         findInPageButton.snp.remakeConstraints { (make) in
             make.leading.trailing.equalTo(safeAreaLayoutGuide)
-            if searchSuggestionsVisible > 0 {
-                make.top.equalTo(searchButtonGroup[searchSuggestionsVisible-1].snp.bottom)
+            if lastSearchButtonIndex >= 0 && !searchButtonGroup[lastSearchButtonIndex].isHidden {
+                make.top.equalTo(searchButtonGroup[lastSearchButtonIndex].snp.bottom)
             } else {
                 make.top.equalTo(topBorder.snp.bottom)
             }
             make.height.equalTo(UIConstants.layout.overlayButtonHeight)
         }
-        layoutIfNeeded()
+        
+        self.setAttributedButtonTitle(phrase: self.searchQuery, button: self.findInPageButton, localizedStringFormat: UIConstants.strings.findInPageButton)
     }
 
-    fileprivate func updateCopyConstraint(showCopyButton: Bool) {
-        if showCopyButton {
-            copyButton.isHidden = false
-            if searchButtonGroup[0].isHidden || searchQueryArray[0].isEmpty {
-                copyButton.snp.remakeConstraints { make in
-                    if searchSuggestionsPrompt.isHidden {
-                        make.top.leading.trailing.equalTo(safeAreaLayoutGuide)
-                    } else {
-                        make.leading.trailing.equalTo(safeAreaLayoutGuide)
-                        make.top.equalTo(searchSuggestionsPrompt.snp.bottom)
-                    }
-                    make.height.equalTo(UIConstants.layout.overlayButtonHeight)
-                }
-            } else if findInPageButton.isHidden {
-                copyButton.snp.remakeConstraints { make in
-                    make.leading.trailing.equalTo(safeAreaLayoutGuide)
-                    make.top.equalTo(searchButtonGroup[searchSuggestionsVisible-1].snp.bottom)
-                    make.height.equalTo(UIConstants.layout.overlayButtonHeight)
-                }
-            } else {
-                copyButton.snp.remakeConstraints { make in
-                    make.leading.trailing.equalTo(safeAreaLayoutGuide)
+    fileprivate func updateCopyConstraints(isCopyButtonHidden: Bool, isFindInPageHidden: Bool, lastSearchButtonIndex: Int) {
+        copyButton.isHidden = isCopyButtonHidden
+        
+        if !isCopyButtonHidden {
```

Comment:
Better to use an early return here instead of everything being inside an if statement:

```if isCopyButtonHidden { return }```